File: Library_API/BusinessLogicLayer/DTOs/BookCopyCreateRequest.cs
using DataAccessLayer.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusinessLogicLayer.DTOs
{
    public record BookCopyCreateRequest(string Title,
        string ISBN,
        string Publisher,
        int Year,
        int PagesCount,
        string Description,
        string CoverImageUrl,
        BookStatus Status,
        List<string> AuthorNames,
        List<string> GenreNames,
        List<string> LibraryNames);
}

File: Library_API/UI/library-app/src/app/interfaces/book-copy-create-request.ts
export interface BookCopyRequest {
    title: string,
    isbn: string,
    publisher: string,
    year: number,
    pagesCount: number,
    description: string,
    coverImageUrl: string,
    authorNames: string[],
    genreNames: string[],
    status: string,
    libraryNames: string[]
}
